services/jira/jira-sprint.service.js

```diff
@@ -33,74 +56,57 @@ module.exports = class JiraSprint extends LegacyService {
-        const sprintId = match[3] // eg, 94
-        const format = match[4] // eg, png
-
-        const options = {
-          method: 'GET',
-          json: true,
-          uri: `${protocol}://${host}/rest/api/2/search?jql=sprint=${sprintId}%20AND%20type%20IN%20(Bug,Improvement,Story,"Technical%20task")&fields=resolution&maxResults=500`,
-        }
-        if (serverSecrets && serverSecrets.jira_username) {
-          options.auth = {
-            user: serverSecrets.jira_username,
-            pass: serverSecrets.jira_password,
-          }
-        }
+  async handle({ protocol, host, path, sprintId }) {
+    let url = `${protocol}://${host}`
+    if (path) {
+      url += `/${path}`
+    }
+    // Atlassian Documentation: https://developer.atlassian.com/cloud/jira/platform/rest/v2/#api-group-Search
+    // There are other sprint-specific APIs but those require authentication. The search API
+    // allows us to get the needed data without being forced to authenticate.
+    url += '/rest/api/2/search'
+    const options = {
+      qs: {
+        jql: `sprint=${sprintId} AND type IN (Bug,Improvement,Story,"Technical task")`,
+        fields: 'resolution',
+        maxResults: 500,
+      },
+    }
+    const json = await this.fetch({
+      url,
+      schema,
+      options,
+      errorMessages: {
+        400: 'sprint not found',
+        404: 'sprint not found',
+      },
+    })
+    const numTotalIssues = json.total
+    const numCompletedIssues = json.issues.filter(issue => {
+      if (issue.fields.resolution != null) {
+        return issue.fields.resolution.name !== 'Unresolved'
+      }
+    }).length
 
-        const badgeData = getBadgeData('completion', data)
-        request(options, (err, res, json) => {
-          if (err != null) {
-            badgeData.text[1] = 'inaccessible'
-            sendBadge(format, badgeData)
-            return
-          }
-          try {
-            if (json && json.total >= 0) {
-              const issuesDone = json.issues.filter(el => {
-                if (el.fields.resolution != null) {
-                  return el.fields.resolution.name !== 'Unresolved'
-                }
-              }).length
-              badgeData.text[1] = `${Math.round(
-                (issuesDone * 100) / json.total
-              )}%`
-              switch (issuesDone) {
-                case 0:
-                  badgeData.colorscheme = 'red'
-                  break
-                case json.total:
-                  badgeData.colorscheme = 'brightgreen'
-                  break
-                default:
-                  badgeData.colorscheme = 'orange'
-              }
-            } else {
-              badgeData.text[1] = 'invalid'
-            }
-            sendBadge(format, badgeData)
-          } catch (e) {
-            badgeData.text[1] = 'invalid'
-            sendBadge(format, badgeData)
-          }
-        })
-      })
-    )
+    const percentComplete = numTotalIssues
```

Comment:
From this point down you've moved on to rendering. Could you move this into `render()`?